aten/src/ATen/native/cudnn/MHA.cpp

```diff
@@ -614,9 +664,44 @@ void run_cudnn_SDP_bprop(
     Tensor& dV,
     const Tensor& dropoutseed,
     const Tensor& dropoutoffset) {
+  if (cudnn_sdpa_check_debug()) {
+    TORCH_WARN("Calling into cuDNN SDPA BPROP");
+  }
+  Tensor dO_ = dO;
+  if (!dO.strides()[dO.strides().size() - 1]) {
+    TORCH_WARN(
+        "cuDNN SDPA backward got an innermost stride of 0 in grad_out, which is unsupported."
+        " Materializing a contiguous tensor which will increase memory usage...");
+    dO_ = dO.contiguous();
+  }
+  if (!std::equal(
+      o.strides().begin(), o.strides().end(), dO.strides().begin())) {
+    TORCH_WARN(
```

Comment:
kinda same here

Reply:
This is actually what causes the failing `causal_variant` test, the `grad_output` used by the test comes from another backend which may not match the memory layout produced by `cudnn_attention`, and cuDNN seems to just assume `grad_output` has the same stride as `output`. This warning I consider a nag to get cuDNN to properly support `grad_output` having different strides from `output`.